colpali_engine/trainer/retrieval_evaluator.py

```diff
@@ -2,6 +2,14 @@
 from mteb.evaluation.evaluators import RetrievalEvaluator
 
 
+if torch.cuda.is_available():
```

Comment:
I'd suggest adding the following function in `utils/` and importing it here.

**`utils/torch_utils.py`**:

```python
import torch


def get_torch_device(device: str = "auto") -> str:
    """
    Returns the device (string) to be used by PyTorch.

    Defaults to "auto" which will use:
    - "cuda:0" if available
    - else "mps" if available
    - "cpu" otherwise.
    """
    if device == "auto":
        if torch.cuda.is_available():
            return "cuda:0"
        elif torch.backends.mps.is_available():  # for Apple Silicon
            return "mps"
        else:
            return "cpu"
    else:
        return device
```

Then, could you add a `device: str = "auto"` arg to `CustomEvaluator` please? Finally, if you add:

```python
self.device = get_torch_device(device)
```

and replace your added `device` with `self.device`, we should be good for a merge!

Reply:
Woops sorry, a bit redundant with @ManuelFay's comment but glad to see we shared the same opinion ahah